subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -662,6 +663,9 @@ int nrf_cloud_pgps_find_prediction(struct nrf_cloud_pgps_prediction **prediction
 		return -ETIMEUNKNOWN;
 	} else if (cur_gps_sec > end_sec) {
 		if ((cur_gps_sec - end_sec) > PGPS_MARGIN_SEC) {
+			index.cur_pnum = 0xff;
+			state = PGPS_EXPIRED;
+			pgps_need_assistance = false; /* make sure we request it */
```

Comment:
should the `atomic_` functions be used to modify this?

Reply:
`pgps_need_assistance = false;` reads like assistance is not needed.
and the comment reads like assistance IS needed.

Reply:
changed the name; also fixed an issue that caused double ephemeris injections

Reply:
are the `atomic_` functions not required when modifying the `atomic_t` variables?

Reply:
No, not when they are only set to a fixed value and compared to being non-zero -- when they are used as a simple boolean flag.  In fact, using `atomic_t` is overkill for such a usecase. It is definitely needed when manipulating bits within an integer from multiple threads, or when needing to to test-and-set or other interruptable operations.

Setting an atomic_t to a single value (false or true) is an atomic operation already. Checking if an atomic_t value is 0 or not is also an atomic operation.

`typedef long atomic_t;`

Reply:
ok so using `atomic_t` has no benefit in this case.

Reply:
I've changed them to be volatile bool variables, but in  the PR stacked on top of this one.